binding.cc

```diff
@@ -1179,6 +1180,26 @@ namespace zmq {
     NanReturnValue(NanNew<String>(version_info));
   }
 
+#if ZMQ_VERSION_MAJOR >= 4
+   static NAN_METHOD(ZmqCurveKeypair) {
+    NanScope();
+
+    char public_key [41];
+    char secret_key [41];
+
+    int rc = zmq_curve_keypair( public_key, secret_key);
+    if (rc < 0) {
+      throw std::runtime_error(ErrorMessage());
```

Comment:
sure, this is all around line `1192` on your diff here, the [case where `rc` is not zero](https://github.com/zeromq/libzmq/blob/f03a78bbfc205e12591a256914c6d53cc57e9023/doc/zmq_curve_keypair.txt#L25)